Build and deploy HTML on GitHub Actions

hugovk opened the pull request:
(Split from https://github.com/python/peps/pull/1570.)

<!--

Please include the PEP number in the pull request title, example:

PEP NNN: Summary of the changes made

In addition, please sign the CLA.

For more information, please read our Contributing Guidelines (CONTRIBUTING.rst)

-->

Adds a GitHub Actions workflow to build and deploy, like Travis CI.

Notes:

* Travis CI builds against 3.7 and 3.7-dev. I only added 3.8 to GHA. Should it build against a range? For example, we can do 3.6-3.9.

* It will only deploy for `python/peps` and `master` (like Travis)

* I've left Travis CI in place for now. I'd suggest confirming the deploy works first from GHA, then removing from Travis in a followup PR.

* (I also noticed that on Travis, it's deploying for both 3.7 and 3.7-dev. It probably only needs doing from a single job.)

* Nice bonus: `make -j$(nproc)` is quicker on GHA (~80s) than Travis (~110s)

## TODO

* [ ] The deploy will need credentials. I guess there are some at 
https://travis-ci.com/github/python/peps? They will need to be added at https://github.com/python/peps/settings/secrets

pradyunsg commented:
I'm guessing that @ewdurbin would know?

brettcannon commented:
I'm happy to take this if @ewdurbin can get us the secret necessary for the upload and what envvar to set it in.

ewdurbin commented:
<img width="456" alt="Screen Shot 2020-10-21 at 4 14 56 PM" src="https://user-images.githubusercontent.com/1200832/96777670-9cc50580-13b8-11eb-91ed-0262b78845cf.png">

pradyunsg commented:
Awesome, so merging this should be enough to get stuff working? Yay!

brettcannon commented:
@pradyunsg not quite. While Ernest was nice enough to store the secrets, the workflow needs to be updated to actually set those values as environment variables.

hugovk commented:
Ah yes, updated!

# Demo

I added some dummy values in my fork. With another branch, before setting env vars, fails to find credentials (as expected):

```
upload failed: build/peps.tar.gz to s3://pythondotorg-assets-staging/peps.tar.gz Unable to locate credentials
```

https://github.com/hugovk/peps/runs/1294073460?check_suite_focus=true#step:6:52

After adding env vars, found them (and dummy values didn't work, as expected):

```
upload failed: build/peps.tar.gz to s3://pythondotorg-assets-staging/peps.tar.gz An error occurred (InvalidAccessKeyId) when calling the PutObject operation: The AWS Access Key Id you provided does not exist in our records.
```

https://github.com/hugovk/peps/runs/1294144579?check_suite_focus=true#step:6:54

brettcannon commented:
I just merged this and the first run worked! So I have gone ahead and removed the Travis webhook and configuration file.

Thanks, @hugovk

hugovk commented:
Great, thanks!
